### PR TITLE
Add automatic deploy to PyPI GitHub Action

### PR DESCRIPTION
**Pull request by sethmlarson**

### CODE REVIEW

File: .github/workflows/deploy.yml
Comment (davisagli) [inline .github/workflows/deploy.yml:5]: Have you configured a tag protection rule for this pattern? https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/configuring-tag-protection-rules

Comment (sethmlarson) [reply .github/workflows/deploy.yml:5]: I have a rule for `v*`, but regardless of this the deploy won't happen without an approval of the `Deploy` environment. It's not super obvious that that's been configured in the GitHub Action config itself, we'll have to test it out once we merge this.

Comment (davisagli) [reply .github/workflows/deploy.yml:5]: Ah yes, I overlooked `environment: Deploy`. I haven't used environments yet but am eager to see how it works.

**Review by davisagli — APPROVED** [review]

**Review by davisagli — APPROVED** [review]

**Pull request merged**
